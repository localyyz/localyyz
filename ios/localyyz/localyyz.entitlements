<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aps-environment</key>
	<string>development</string>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>applinks:localyyz.app.link</string>
		<string>applinks:localyyz-alternate.app.link</string>
		<string>applinks:localyyz.test-app.link</string>
		<string>applinks:localyyz-alternate.test-app.link</string>
	</array>
	<key>com.apple.developer.in-app-payments</key>
	<array>
		<string>merchant.com.localyyz.toronto</string>
	</array>
</dict>
</plist>
