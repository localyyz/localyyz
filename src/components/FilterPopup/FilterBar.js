import React from "react";
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Text,
  FlatList
} from "react-native";
import PropTypes from "prop-types";

// custom
import { Styles, Colours, Sizes } from "localyyz/constants";
import { Filter, SloppyView } from "localyyz/components";
import { withCommas } from "localyyz/helpers";

// third party
import { observer, inject } from "mobx-react/native";
import { withNavigation } from "react-navigation";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";

// local
import Popup from "./Popup";

// constants
const SORT_BY = [
  { label: "What's new", value: "created_at" },
  { label: "Price", value: "price", hasDirection: true },
  { label: "Discount (% off)", value: "discount" }
];

@inject(stores => ({
  store: stores.filterStore,
  numProducts: Math.max(
    0,
    stores.productListStore.numProducts || 0,
    stores.productListStore.products.length
  )
}))
@observer
export class FilterBar extends React.Component {
  static propTypes = {
    isInitialVisible: PropTypes.bool,
    onPress: PropTypes.func
  };

  static defaultProps = {
    text: "Refine",
    isInitialVisible: false
  };

  constructor(props) {
    super(props);

    // bindings
    this.openFilter = this.openFilter.bind(this);
    this.renderItem = this.renderItem.bind(this);
  }

  componentDidMount() {
    this.props.isInitialVisible ? this.onPress() : null;
  }

  openFilter() {
    this.props.onPress
      ? this.props.onPress()
      : this.props.navigation.navigate("Modal", {
          component: <Popup {...this.props} />
        });
  }

  renderItem({ item: sorter }) {
    return <SortOption {...sorter} />;
  }

  get genderFilter() {
    return !this.props.hideGenderFilter ? (
      <View style={{ marginLeft: Sizes.InnerFrame }}>
        <View style={styles.circle}>
          <Filter.Gender />
        </View>
      </View>
    ) : null;
  }

  get header() {
    return (
      <View style={Styles.Horizontal}>
        <TouchableOpacity onPress={this.openFilter}>
          <SloppyView>
            <View style={styles.circle}>
              <MaterialIcon
                name="sort"
                size={Sizes.Text}
                color={Colours.Text}/>
            </View>
          </SloppyView>
        </TouchableOpacity>
        {this.genderFilter}
        <View style={styles.optionContainer}>
          <Text style={styles.optionHeader}>Sort by</Text>
        </View>
      </View>
    );
  }

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          horizontal
          showsHorizontalScrollIndicator={false}
          data={SORT_BY}
          extraData={this.props.sortBy}
          keyExtractor={(option, i) => `option${i}-${option.value}`}
          contentContainerStyle={styles.content}
          ListHeaderComponent={this.header}
          renderItem={this.renderItem}/>
        <Text style={styles.productCount}>{`${withCommas(
          this.props.numProducts
        )} products`}</Text>
      </View>
    );
  }
}

@inject(stores => ({
  sortBy: stores.filterStore.sortBy,
  setSortBy: stores.filterStore.setSortBy
}))
@observer
class SortOption extends React.Component {
  constructor(props) {
    super(props);

    // bindings
    this.onPress = this.onPress.bind(this);
  }

  get isSelected() {
    let parts = this.props.sortBy ? this.props.sortBy.split("-") : [];
    return parts.length > 0 && parts[parts.length - 1] === this.props.value;
  }

  get isDescending() {
    return (
      this.props.sortBy
      && this.props.sortBy.length > 0
      && this.props.sortBy[0] === "-"
    );
  }

  onPress() {
    if (!this.isSelected) {
      this.props.setSortBy(this.props.value);
    } else if (this.props.hasDirection) {
      this.props.setSortBy(!this.isDescending ? `-${this.props.value}` : null);
    } else {
      this.props.setSortBy();
    }
  }

  render() {
    return (
      <View style={styles.optionContainer}>
        <TouchableOpacity onPress={this.onPress}>
          <SloppyView>
            <View
              style={[styles.option, this.isSelected && styles.selectedOption]}>
              <Text
                style={[
                  styles.optionLabel,
                  styles.optionLabel,
                  this.isSelected && styles.selectedOptionLabel
                ]}>
                {this.props.label}
              </Text>
              {this.props.hasDirection && this.isSelected ? (
                <View style={styles.optionDirection}>
                  <MaterialIcon
                    name={this.isDescending ? "arrow-downward" : "arrow-upward"}
                    size={Sizes.TinyText}/>
                </View>
              ) : null}
            </View>
          </SloppyView>
        </TouchableOpacity>
      </View>
    );
  }
}

export default withNavigation(FilterBar);

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colours.Foreground
  },

  content: {
    alignItems: "center",
    paddingHorizontal: Sizes.InnerFrame
  },

  optionContainer: {
    marginLeft: Sizes.InnerFrame / 2
  },

  option: {
    ...Styles.Horizontal,
    ...Styles.RoundedButton,
    marginTop: 1,
    paddingHorizontal: Sizes.InnerFrame / 4,
    paddingVertical: Sizes.InnerFrame / 4,
    backgroundColor: Colours.Transparent
  },

  optionLabel: {
    ...Styles.Text,
    ...Styles.Emphasized,
    ...Styles.SmallText
  },

  optionDirection: {
    marginLeft: Sizes.InnerFrame / 4
  },

  optionHeader: {
    ...Styles.Subdued,
    marginLeft: Sizes.InnerFrame / 2
  },

  selectedOption: {
    backgroundColor: Colours.Action,
    paddingHorizontal: Sizes.InnerFrame
  },

  selectedOptionLabel: {},

  productCount: {
    ...Styles.Text,
    ...Styles.SmallText,
    ...Styles.Emphasized,
    ...Styles.Subdued,
    paddingTop: Sizes.InnerFrame / 2,
    paddingHorizontal: Sizes.InnerFrame
  },

  circle: {
    width: Sizes.InnerFrame * 2,
    height: Sizes.InnerFrame * 2,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: Sizes.InnerFrame,
    backgroundColor: Colours.Action
  }
});
