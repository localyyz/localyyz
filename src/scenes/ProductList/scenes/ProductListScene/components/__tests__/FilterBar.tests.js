import React from "react";

// custom
import { findById } from "localyyz/tests";
import ShallowRenderer from "react-test-renderer/shallow";
import Renderer from "react-test-renderer";

// local
import { FilterBar, SortOption } from "../FilterBar";
let Component = FilterBar.wrappedComponent;
let SortOptionComponent = SortOption.wrappedComponent;
let shallowRenderer = new ShallowRenderer();

// constants
const NAME = "FilterBar";
const PROPS = {
  store: {},
  numProducts: 100
};

const SO_NAME = "SortOption";
const SO_PROPS = {
  label: "What's new",
  value: "created_at"
};

// helpers
function getHeader(rendered) {
  return rendered.props.children[0].props.ListHeaderComponent;
}

describe(NAME, () => {
  it("should render properly", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const c = shallowRenderer.getRenderOutput();

    expect(c).toBeDefined();
  });

  it("should show no filters active", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "brand")).not.toBeDefined();
    expect(findById(c, "price")).not.toBeDefined();
    expect(findById(c, "discount")).not.toBeDefined();
    expect(findById(c, "color")).not.toBeDefined();
    expect(findById(c, "size")).not.toBeDefined();
  });

  it("should show price filter", () => {
    shallowRenderer.render(<Component {...PROPS} store={{ priceMax: 100 }} />);
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "price")).toBeDefined();
    expect(c).toMatchSnapshot();
  });

  it("should show brand filter", () => {
    shallowRenderer.render(
      <Component {...PROPS} store={{ brand: "supreme" }} />
    );
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "brand")).toBeDefined();
    expect(c).toMatchSnapshot();
  });

  it("should show discount filter", () => {
    shallowRenderer.render(
      <Component {...PROPS} store={{ discountMin: 0.5 }} />
    );
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "discount")).toBeDefined();
    expect(c).toMatchSnapshot();
  });

  it("should show color filter", () => {
    shallowRenderer.render(<Component {...PROPS} store={{ color: "black" }} />);
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "color")).toBeDefined();
    expect(c).toMatchSnapshot();
  });

  it("should show size filter", () => {
    shallowRenderer.render(<Component {...PROPS} store={{ size: "large" }} />);
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "size")).toBeDefined();
    expect(c).toMatchSnapshot();
  });

  it("should show gender filter when no gender is given", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "gender")).toBeDefined();
    expect(c).toMatchSnapshot();
  });

  it("should hide gender filter", () => {
    shallowRenderer.render(<Component {...PROPS} hideGenderFilter={true} />);
    const c = getHeader(shallowRenderer.getRenderOutput());

    expect(findById(c, "gender")).not.toBeDefined();
  });

  it("should render to snapshot", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe(SO_NAME, () => {
  it("should render properly", () => {
    let render = Renderer.create(
      <SortOptionComponent {...SO_PROPS} sortBy={SO_PROPS.value} />
    ).toJSON();

    expect(render).toBeDefined();
  });

  it("should render to snapshot", () => {
    let render = Renderer.create(
      <SortOptionComponent {...SO_PROPS} sortBy={SO_PROPS.value} />
    ).toJSON();

    expect(render).toMatchSnapshot();
  });

  it("should be highlighted", () => {
    let render = Renderer.create(
      <SortOptionComponent {...SO_PROPS} sortBy={SO_PROPS.value} />
    ).getInstance();
    expect(render.isSelected).toBeTruthy();
  });

  it("should be not be highlighted", () => {
    let render = Renderer.create(
      <SortOptionComponent {...SO_PROPS} />
    ).getInstance();
    expect(render.isSelected).toBeFalsy();
  });

  it("should set on toggle if unset", () => {
    let setSortBy = jest.fn();
    let render = Renderer.create(
      <SortOptionComponent {...SO_PROPS} setSortBy={setSortBy} />
    ).getInstance();

    // press the button
    render.onPress();
    expect(setSortBy.mock.calls[0][0]).toBe(SO_PROPS.value);
  });

  it("should unset on toggle if set", () => {
    let setSortBy = jest.fn();
    let render = Renderer.create(
      <SortOptionComponent
        {...SO_PROPS}
        sortBy={SO_PROPS.value}
        setSortBy={setSortBy}/>
    ).getInstance();

    // press the button
    render.onPress();
    expect(setSortBy.mock.calls[0][0]).toEqual(undefined);
  });

  it("should reverse direction on toggle if has direction and is set", () => {
    let setSortBy = jest.fn();
    let render = Renderer.create(
      <SortOptionComponent
        {...SO_PROPS}
        hasDirection
        sortBy={SO_PROPS.value}
        setSortBy={setSortBy}/>
    ).getInstance();

    // press the button
    render.onPress();
    expect(setSortBy.mock.calls[0][0]).toBe(`-${SO_PROPS.value}`);
  });

  it("should unset on toggle if has direction and is set to descending", () => {
    let setSortBy = jest.fn();
    let render = Renderer.create(
      <SortOptionComponent
        {...SO_PROPS}
        hasDirection
        sortBy={`-${SO_PROPS.value}`}
        setSortBy={setSortBy}/>
    ).getInstance();

    // press the button
    render.onPress();
    expect(setSortBy.mock.calls[0][0]).toEqual(null);
  });
});
