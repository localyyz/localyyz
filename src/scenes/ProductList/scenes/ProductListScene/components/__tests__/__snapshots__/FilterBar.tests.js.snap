// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should render to snapshot 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
  testID="hello"
>
  <FlatList
    ListHeaderComponent={
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Component
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
              }
            }
          >
            Filter by
          </Text>
        </Component>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <SloppyView>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#DEEFF5",
                  "borderRadius": 15,
                  "height": 30,
                  "justifyContent": "center",
                  "marginHorizontal": 7.5,
                  "width": 30,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                color="rgb(30, 30, 30)"
                name="playlist-add"
                size={16}
              />
            </Component>
          </SloppyView>
        </TouchableOpacity>
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DEEFF5",
              "borderRadius": 15,
              "height": 30,
              "justifyContent": "center",
              "marginHorizontal": 7.5,
              "width": 30,
            }
          }
          testID="gender"
        >
          <inject-Gender />
        </Component>
        <Component
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
              }
            }
          >
            Sort by
          </Text>
        </Component>
      </Component>
    }
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingHorizontal": 15,
      }
    }
    data={
      Array [
        Object {
          "label": "What's new",
          "value": "created_at",
        },
        Object {
          "hasDirection": true,
          "label": "Price",
          "value": "price",
        },
        Object {
          "label": "Discount (% off)",
          "value": "discount",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0)",
        "color": "#999999",
        "fontSize": 14,
        "fontWeight": "600",
        "paddingHorizontal": 15,
        "paddingTop": 7.5,
      }
    }
  >
    100 products
  </Text>
</Component>
`;

exports[`FilterBar should show brand filter 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Filter by
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <SloppyView>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "borderRadius": 15,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 1,
                "paddingHorizontal": 3.75,
                "paddingVertical": 3.75,
              },
              Object {
                "backgroundColor": "#DEEFF5",
                "marginHorizontal": 3.75,
                "paddingHorizontal": 15,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
            testID="brand"
          >
            Only Supreme
          </Text>
        </Component>
      </SloppyView>
    </TouchableOpacity>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <SloppyView>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DEEFF5",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 7.5,
            "width": 30,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(30, 30, 30)"
          name="playlist-add"
          size={16}
        />
      </Component>
    </SloppyView>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DEEFF5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginHorizontal": 7.5,
        "width": 30,
      }
    }
    testID="gender"
  >
    <inject-Gender />
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Sort by
    </Text>
  </Component>
</Component>
`;

exports[`FilterBar should show color filter 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Filter by
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <SloppyView>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "borderRadius": 15,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 1,
                "paddingHorizontal": 3.75,
                "paddingVertical": 3.75,
              },
              Object {
                "backgroundColor": "#DEEFF5",
                "marginHorizontal": 3.75,
                "paddingHorizontal": 15,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
            testID="color"
          >
            Only Black
          </Text>
        </Component>
      </SloppyView>
    </TouchableOpacity>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <SloppyView>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DEEFF5",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 7.5,
            "width": 30,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(30, 30, 30)"
          name="playlist-add"
          size={16}
        />
      </Component>
    </SloppyView>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DEEFF5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginHorizontal": 7.5,
        "width": 30,
      }
    }
    testID="gender"
  >
    <inject-Gender />
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Sort by
    </Text>
  </Component>
</Component>
`;

exports[`FilterBar should show discount filter 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Filter by
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <SloppyView>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "borderRadius": 15,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 1,
                "paddingHorizontal": 3.75,
                "paddingVertical": 3.75,
              },
              Object {
                "backgroundColor": "#DEEFF5",
                "marginHorizontal": 3.75,
                "paddingHorizontal": 15,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
            testID="discount"
          >
            At Least 50% Off
          </Text>
        </Component>
      </SloppyView>
    </TouchableOpacity>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <SloppyView>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DEEFF5",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 7.5,
            "width": 30,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(30, 30, 30)"
          name="playlist-add"
          size={16}
        />
      </Component>
    </SloppyView>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DEEFF5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginHorizontal": 7.5,
        "width": 30,
      }
    }
    testID="gender"
  >
    <inject-Gender />
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Sort by
    </Text>
  </Component>
</Component>
`;

exports[`FilterBar should show gender filter when no gender is given 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Filter by
    </Text>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <SloppyView>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DEEFF5",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 7.5,
            "width": 30,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(30, 30, 30)"
          name="playlist-add"
          size={16}
        />
      </Component>
    </SloppyView>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DEEFF5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginHorizontal": 7.5,
        "width": 30,
      }
    }
    testID="gender"
  >
    <inject-Gender />
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Sort by
    </Text>
  </Component>
</Component>
`;

exports[`FilterBar should show price filter 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Filter by
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <SloppyView>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "borderRadius": 15,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 1,
                "paddingHorizontal": 3.75,
                "paddingVertical": 3.75,
              },
              Object {
                "backgroundColor": "#DEEFF5",
                "marginHorizontal": 3.75,
                "paddingHorizontal": 15,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
            testID="price"
          >
            Under $100
          </Text>
        </Component>
      </SloppyView>
    </TouchableOpacity>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <SloppyView>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DEEFF5",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 7.5,
            "width": 30,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(30, 30, 30)"
          name="playlist-add"
          size={16}
        />
      </Component>
    </SloppyView>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DEEFF5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginHorizontal": 7.5,
        "width": 30,
      }
    }
    testID="gender"
  >
    <inject-Gender />
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Sort by
    </Text>
  </Component>
</Component>
`;

exports[`FilterBar should show size filter 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Filter by
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <SloppyView>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "borderRadius": 15,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 1,
                "paddingHorizontal": 3.75,
                "paddingVertical": 3.75,
              },
              Object {
                "backgroundColor": "#DEEFF5",
                "marginHorizontal": 3.75,
                "paddingHorizontal": 15,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
            testID="size"
          >
            Only Large
          </Text>
        </Component>
      </SloppyView>
    </TouchableOpacity>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <SloppyView>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DEEFF5",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 7.5,
            "width": 30,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="rgb(30, 30, 30)"
          name="playlist-add"
          size={16}
        />
      </Component>
    </SloppyView>
  </TouchableOpacity>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DEEFF5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginHorizontal": 7.5,
        "width": 30,
      }
    }
    testID="gender"
  >
    <inject-Gender />
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999999",
        }
      }
    >
      Sort by
    </Text>
  </Component>
</Component>
`;

exports[`SortOption should render to snapshot 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "borderRadius": 15,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 1,
              "paddingHorizontal": 3.75,
              "paddingVertical": 3.75,
            },
            Object {
              "backgroundColor": "#DEEFF5",
              "marginHorizontal": 3.75,
              "paddingHorizontal": 15,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {},
            ]
          }
        >
          What's New
        </Text>
      </View>
    </View>
  </View>
</View>
`;
